admin_delete_test.go

```diff
@@ -0,0 +1,62 @@
+package cloudinary
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBulkDeleteUploadedWithInvalidation(t *testing.T) {
+	testCases := []struct {
+		name string
+		publicIDs []string
+	}{
+		{
+			name: "0 ids should work ok",
+			publicIDs: []string{},
+		},
+		{
+			name: "Less than 20 ids should work ok",
+			publicIDs: []string{"1", "2", "3"},
+		},
+		{
+			name: "20 ids should work ok",
+			publicIDs: []string{"1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20"},
+		},
+		{
+			name: "More than 20 ids should work ok",
+			publicIDs: []string{"1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21"},
+		},
+	}
+
+	s := &Service{}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := s.bulkDeleteUploadedWithInvalidation(&mockHttpClient{}, ImageType, tc.publicIDs)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+type mockHttpClient struct {}
```

Comment:
Please use mock.Mock as we did all around the project.

Reply:
fixed

Reply:
replaced mock by httptest.
So changes in the method itself are minimal (no client dependency is needed anymore).